addons/project/models/project.py

```diff
@@ -1604,7 +1604,16 @@ def filtered_domain(self, domain):
 
     def copy_data(self, default=None):
         defaults = super().copy_data(default=default)
-        return [{k: v for k, v in default.items() if k in self.SELF_READABLE_FIELDS} for default in defaults]
+        extra_fields = []
+        if not self.env.user.has_group('base.group_portal'):
+            custom_fields = set()
+            for default in defaults:
+                custom_fields |= set(filter(lambda k: k.startswith('x_'), default.keys()))
+            extra_fields = self.env['ir.model.fields'].sudo().search([
+                ('name', 'in', list(custom_fields)),
+                ('copied', '=', True)
+            ]).mapped('name')
+        return [{k: v for k, v in default.items() if k in self.SELF_READABLE_FIELDS or k in extra_fields} for default in defaults]
```

Comment:
Alternative solution:

```suggestion
        if not self.env.user.has_group('base.group_portal'):
            return defaults
        return [{k: v for k, v in default.items() if k in self.SELF_READABLE_FIELDS} for default in defaults]
```

Reply:
Thanks for the fix but I prefer the alternative one that you suggested. 
However, we could check if the current user is a Project User (`project.group_project_user`) instead of checking if he is not a Portal one. So, we can be sure the current user has access to the Project app. 🙂